internal/common/utils/utils.go

```diff
@@ -122,3 +122,17 @@ func ExtractInfoFromEnv(env []corev1.EnvVar, configmapMap MapType, ref GetRefEnv
 		}
 	}
 }
+
+// FilterByPodmanPrefix does two things:
+// first it filters the keys that start with prefix "podman/"
+// And secondly, of those keys that have the "podman/" prefix, it uses the remaining part of the key as the new key in the returning map with the same value as in the original map
+// e.g. 'podman/this-is-my-key:1' is stored in the returning map as 'this-is-my-key:1'
+func FilterByPodmanPrefix(keyValues map[string]string) map[string]string {
+	ret := map[string]string{}
```

Comment:
why not `make(map[string]string)` ? just a question

Reply:
Personal choice, I only use `make` when I need to create a map, channel or slice with a predefined size. Otherwise I keep it consistent with `{}`.

Reply:
ok but here you a size. maximum what you can get is len(keyValues)

Reply:
I pprofed 2 code one with/without make(map[string]string)

without make this is the mem:
```
 1762.94kB 40.77% 40.77%  1762.94kB 40.77%  runtime/pprof.StartCPUProfile
  512.50kB 11.85% 52.63%   512.50kB 11.85%  runtime.allocm
  512.20kB 11.85% 64.47%   512.20kB 11.85%  runtime.malg
  512.05kB 11.84% 76.32%   512.05kB 11.84%  regexp/syntax.(*parser).newRegexp (inline)
  512.04kB 11.84% 88.16%   512.04kB 11.84%  runtime.bgscavenge
  512.02kB 11.84%   100%   512.02kB 11.84%  runtime.gcBgMarkWorker
         0     0%   100%   512.05kB 11.84%  internal/profile.init
         0     0%   100%  1762.94kB 40.77%  main.main
         0     0%   100%   512.05kB 11.84%  regexp.Compile (inline)
         0     0%   100%   512.05kB 11.84%  regexp.MustCompile
```

with make:
``` 
1184.27kB 36.63% 36.63%  1184.27kB 36.63%  runtime/pprof.StartCPUProfile
 1024.41kB 31.68% 68.31%  1024.41kB 31.68%  runtime.malg
  512.50kB 15.85% 84.16%   512.50kB 15.85%  runtime.allocm
  512.02kB 15.84%   100%   512.02kB 15.84%  runtime.gcBgMarkWorker
         0     0%   100%  1184.27kB 36.63%  main.main
         0     0%   100%  1184.27kB 36.63%  runtime.main
         0     0%   100%   512.50kB 15.85%  runtime.mstart
         0     0%   100%   512.50kB 15.85%  runtime.mstart0
         0     0%   100%   512.50kB 15.85%  runtime.mstart1
         0     0%   100%   512.50kB 15.85%  runtime.newm
```

If you look at `main` in the case with you allocated less memory. I think is related to how maps are layout in memory.

Reply:
already merged :)

Reply:
No worries, I can raise a new one for this, but thanks a lot for taking the time to check this out! :smile:

Reply:
this was purely theoretical. don't worry